jsonnet/components/metrics-server.libsonnet

```diff
@@ -0,0 +1,377 @@
+                  port: 'https',
+                  scheme: 'HTTPS',
+                },
+                periodSeconds: 10,
+              },
+              name: 'metrics-server',
+              ports: [
+                {
+                  containerPort: 10250,
+                  name: 'https',
+                  protocol: 'TCP',
+                },
+              ],
+              readinessProbe: {
+                failureThreshold: 3,
+                httpGet: {
+                  path: '/readyz',
+                  port: 'https',
+                  scheme: 'HTTPS',
+                },
+                initialDelaySeconds: 20,
+                periodSeconds: 10,
+              },
+              resources: {
+                requests: {
+                  cpu: '1m',
+                  memory: '40Mi',
+                },
+              },
+              securityContext: {
+                allowPrivilegeEscalation: false,
+                readOnlyRootFilesystem: true,
+                runAsNonRoot: true,
+              },
+              volumeMounts: [
+                {
+                  mountPath: '/etc/tls/private',
+                  name: 'secret-metrics-server-tls',
+                },
+                {
+                  mountPath: '/etc/tls/metrics-client-certs',
+                  name: 'secret-metrics-client-certs',
+                },
+                {
+                  mountPath: '/etc/tls/kubelet-serving-ca-bundle',
+                  name: 'configmap-kubelet-serving-ca-bundle',
+                },
+              ],
+            },
+          ],
+          nodeSelector: {
+            'kubernetes.io/os': 'linux',
+          },
+          priorityClassName: 'system-cluster-critical',
+          serviceAccountName: 'metrics-server',
+          volumes: [
+            {
+              name: 'secret-metrics-client-certs',
+              secret: {
+                secretName: 'metrics-client-certs',
+              },
+            },
+            {
+              name: 'secret-metrics-server-tls',
+              secret: {
+                secretName: 'metrics-server-tls',
+              },
+            },
+            {
+              configMap: {
+                name: 'kubelet-serving-ca-bundle',
+              },
+              name: 'configmap-kubelet-serving-ca-bundle',
+            },
+          ],
+        },
+      },
+    },
+  },
+  podDisruptionBudget: {
+    apiVersion: 'policy/v1',
+    kind: 'PodDisruptionBudget',
+    metadata: {
+      labels: {
+        'app.kubernetes.io/name': 'metrics-server',
+        'app.kubernetes.io/component': 'metrics-server',
+      } + cfg.commonLabels,
+      name: 'metrics-server',
+      namespace: cfg.namespace,
+    },
+    spec: {
+      minAvailable: 1,
+      selector: {
+        matchLabels: {
+          'app.kubernetes.io/name': 'metrics-server',
+          'app.kubernetes.io/component': 'metrics-server',
+        } + cfg.commonLabels,
+      },
+    },
+  },
+  apiService: {
```

Comment:
same remark here: shouldn't it be moved to cluster-monitoring-operator?

Reply:
Since this depends on service we cannot move this one atleast for now